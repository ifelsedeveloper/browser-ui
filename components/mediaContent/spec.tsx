/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import MediaContent from './index'

describe('mediaContent tests', () => {
  const baseComponent = (props?: object) => (
    <MediaContent id='testMediaContent' {...props} />
  )

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent()
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent({id: 'toughMediaContent'}))
      const assertion = wrapper.find('#toughMediaContent').length
      expect(assertion).toBe(1)
    })
  })

  describe('component behavior', () => {
    it('can have an id', () => {
      const wrapper = shallow(baseComponent({id: 'mediaThing'}))
      const assertion = wrapper.props().id
      expect(assertion).toBe('mediaThing')
    })
  })

  describe('theming', () => {
    it('allows theming the `width` property', () => {
      // TODO
    })

    it('allows theming the `margin` property', () => {
      // TODO
    })
  })
})
