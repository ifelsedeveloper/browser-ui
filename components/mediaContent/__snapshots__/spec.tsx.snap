// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mediaContent tests basic tests matches the snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  font-weight: inherit;
  font-family: inherit;
  color: inherit;
}

.c1 {
  box-sizing: border-box;
  display: block;
  max-width: 100%;
}

.c2 {
  box-sizing: border-box;
  font-weight: inherit;
  font-family: inherit;
  color: inherit;
}

<div
  className="c0"
  id="testMediaContent"
>
  <img
    className="c1"
    src={undefined}
  />
  <div
    className="c2"
  />
</div>
`;
